test/cppsuite/tests/session_microbenchmarks.cpp

```diff
@@ -0,0 +1,171 @@
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "src/util/instruction_counter.h"
+#include "src/common/constants.h"
+#include "src/common/logger.h"
+#include "src/main/test.h"
+
+namespace test_harness {
+
+static void
+make_insert(thread_worker *tc, const std::string &id)
+{
+    std::string cursor_uri = tc->db.get_collection(0).name;
+
+    auto cursor = tc->session.open_scoped_cursor(cursor_uri);
+    cursor->set_key(cursor.get(), "key" + id);
+    cursor->set_value(cursor.get(), "value1" + id);
+    int ret = cursor->insert(cursor.get());
+
+    if (ret != 0) {
+        logger::log_msg(
+          LOG_ERROR, "session_microbenchmarks: expected insert id " + id + " to succeed");
+    }
+}
+
+/*
+ * Class that defines operations that do nothing as an example. This shows how database operations
+ * can be overridden and customized.
+ */
+class session_microbenchmarks : public test {
+public:
+    session_microbenchmarks(const test_args &args) : test(args)
+    {
+        init_operation_tracker(NULL);
+    }
+
+    void
+    checkpoint_operation(thread_worker *) override final
+    {
+        logger::log_msg(LOG_WARN, "checkpoint_operation: not done as this is a performance test");
+    }
+
+    void
+    custom_operation(thread_worker *tc) override final
+    {
+        WT_SESSION wt_session = *(tc->session);
+        instruction_counter begin_transaction_counter(
+          "begin_transaction_instructions", test::_args.test_name);
+        instruction_counter commit_transaction_counter(
+          "commit_transaction_instructions", test::_args.test_name);
+        instruction_counter rollback_trancation_counter(
+          "rollback_trancation_instructions", test::_args.test_name);
+        instruction_counter prepare_transaction_counter(
+          "prepare_transaction_instructions", test::_args.test_name);
+
+        instruction_counter commit_after_prepare_transaction_counter(
+          "commit_after_prepare_instructions", test::_args.test_name);
+        instruction_counter open_cursor_cached_counter(
+          "open_cursor_cached_instructions", test::_args.test_name);
+        instruction_counter open_cursor_uncached_counter(
+          "open_cursor_uncached_instructions", test::_args.test_name);
+        instruction_counter timestamp_transaction_uint_counter(
+          "timestamp_transaction_uint_instructions", test::_args.test_name);
+        std::string cursor_uri = tc->db.get_collection(0).name;
+
+        /* Get the collection to work on. */
+        testutil_assert(tc->collection_count == 1);
+
+        scoped_session &session = tc->session;
+
+        int result = begin_transaction_counter.track([&wt_session, &session]() -> int {
```

Comment:
Seeing as we're using the c++ scoped session we can capture that only and remove the `wt_sessions` var entirely.